src/lib/Containers/Inbox.tsx

```diff
@@ -1,29 +1,47 @@
 import * as React from "react"
 import * as Relay from "react-relay"
+import styled from "styled-components/native"
 
-import { ScrollView } from "react-native"
+import { ScrollView, View } from "react-native"
 import ActiveBids from "../Components/Inbox/Bids"
 import Conversations from "../Components/Inbox/Conversations"
 import ZeroStateInbox from "../Components/Inbox/Conversations/ZerostateInbox"
 
-export class Inbox extends React.Component<any, any> {
-  currentScrollOffset?: number = 0
+interface State {
+  hasBids?: boolean
+  hasMessages?: boolean
+}
+
+const Container = styled.View`
+  flex: 1;
+`
+
+export class Inbox extends React.Component<any, State> {
+  constructor(props) {
+    super(props)
 
-  hasNoContent() {
-    const userHasConversations = this.props.me.conversations.edges.length > 0
-    return !userHasConversations
+    this.state = {
+      hasBids: false,
+      hasMessages: false,
+    }
   }
 
   render() {
-    return this.hasNoContent()
-      ? <ZeroStateInbox />
-      : <ScrollView
-          onScroll={event => (this.currentScrollOffset = event.nativeEvent.contentOffset.y)}
-          scrollEventThrottle={10}
-        >
-          <ActiveBids me={this.props.me} />
-          <Conversations me={this.props.me} />
-        </ScrollView>
+    const updateBidsState = hasBids => {
+      this.setState({ hasBids })
+    }
+    const updateMessagesState = hasMessages => {
+      this.setState({ hasMessages })
+    }
+
+    const shouldShowEmptyState = !this.state.hasBids && !this.state.hasMessages
+    const headerView = <ActiveBids me={this.props.me} onDataLoaded={updateBidsState} />
+    return (
+      <Container>
+        <Conversations me={this.props.me} headerView={headerView} onDataLoaded={updateMessagesState} />
```

Comment:
The result of `Relay.createContainer` _should_ definitely pass though the props of the component, but Relay data is masked by Relay and only visible to the component it belongs to, so you should treat it as an opaque object (and use `any`). Try this:

```tsx
<ActiveBids me={this.props.me as any} onDataLoaded={updateBidState} />
```